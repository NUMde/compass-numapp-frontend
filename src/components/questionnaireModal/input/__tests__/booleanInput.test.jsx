import React from 'react';

import {
  renderWithRedux,
  act,
  fireEvent,
} from '../../../../../__test-utils__/render';
import BooleanInput from '../booleanInput';

describe('BooleanInput', () => {
  it('should render a BooleanInput', () => {
    const { getByTestId, toJSON } = renderWithRedux(
      <BooleanInput
        item={{ linkId: '1.1', type: 'boolean', text: 'booleanInput' }}
      />,
      {
        initialState: {
          Questionnaire: {
            itemMap: {
              1: { linkId: '1', type: 'group', text: 'group' },
              1.1: { linkId: '1.1', type: 'boolean', text: 'boolean' },
            },
          },
        },
      },
    );
    expect(getByTestId('BooleanInput.CheckBox')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should toggle button when clicked', () => {
    const { getByTestId, getByText } = renderWithRedux(
      <BooleanInput
        item={{ linkId: '1.1', type: 'boolean', text: 'booleanInput' }}
      />,
      {
        initialState: {
          Questionnaire: {
            itemMap: {
              1: {
                linkId: '1',
                type: 'group',
                text: 'group',
                item: [{ linkId: '1.1', type: 'boolean', text: 'boolean' }],
              },
              1.1: { linkId: '1.1', type: 'boolean', text: 'boolean' },
            },
            categories: [
              {
                linkId: '1',
                type: 'group',
                text: 'group',
                item: [{ linkId: '1.1', type: 'boolean', text: 'boolean' }],
              },
            ],
          },
        },
      },
    );
    act(() => {
      fireEvent.press(getByTestId('BooleanInput.CheckBox'));
    });
    expect(
      getByTestId('BooleanInput.CheckBox').props.accessibilityState,
    ).toEqual({
      checked: true,
    });
    act(() => {
      fireEvent.press(getByText('\uf046'));
    });
    expect(
      getByTestId('BooleanInput.CheckBox').props.accessibilityState,
    ).toEqual({
      checked: false,
    });
  });
});
