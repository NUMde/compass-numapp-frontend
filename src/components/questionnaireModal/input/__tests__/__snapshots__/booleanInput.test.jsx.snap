// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanInput should render a BooleanInput 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "borderRadius": 3,
      "borderWidth": 1,
      "margin": 0,
      "marginLeft": 0,
      "marginRight": 10,
      "opacity": 1,
      "padding": 10,
      "paddingLeft": 0,
      "paddingRight": 0,
    }
  }
  testID="BooleanInput.CheckBox"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#919191",
            "fontSize": 24,
          },
          Object {
            "minWidth": 24,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï‚–
    </Text>
    <Text
      style={
        Object {
          "color": "#1F2532",
          "fontFamily": "IBMPlexSans-SemiBold",
          "fontSize": 26.25,
          "fontWeight": "bold",
          "lineHeight": 35.625,
          "margin": 0,
          "marginLeft": 10,
          "marginRight": 10,
          "padding": 0,
        }
      }
      testID="checkboxTitle"
    >
      booleanInput
    </Text>
  </View>
</View>
`;
